### DISCUSSION TITLE
Combined sort fields and incorrect sorting

### DISCUSSION THREAD

**Discussion by BigPaul** [microsoft-search/pnp-modern-search#1944] [Q&A]
I'm sorting on two fields:
1. Integer ASC - this field can be empty but is otherwise a unique value
2. Date DESC

If I sort on the integer field alone, the results are as expected no matter the sort direction. There are 8 records with an int value and they appear at the top of the list in the correct order.

If I sort on the date field alone, the results are as expected no matter the sort direction. 

When I combine the two sort fields, with the integer field first, the integer column is sorted as expected at the top of the list, but the date column for the following records is randomly ordered. 

I'm new to SharePoint, and this web-part too off course. Where should I look to find the cause of this issue? 

I can't determine whether this is a bug with the web-part or with my configuration.

Any help much appreciated.

**Comment by wobba** [comment]
First off it helps if you identify which version of the web part you are using and share steps of your web part set up. Sorting on multiple properties is supported by the search API and works like this:

1. Sort on property 1
2. Withing those with same value of property 1, then sort on property 2 (unsure how this affects those not having property 1)

You cannot easily sort on one value, and for the rest another value.
